miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditLibraryTemplateController.java

```diff
@@ -0,0 +1,116 @@
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with MISO. If not, see <http://www.gnu.org/licenses/>.
+ *
+ * *********************************************************************
+ */
+
+package uk.ac.bbsrc.tgac.miso.webapp.controller;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.ModelMap;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.SessionAttributes;
+import org.springframework.web.servlet.ModelAndView;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+
+import uk.ac.bbsrc.tgac.miso.core.data.impl.LibraryTemplate;
+import uk.ac.bbsrc.tgac.miso.dto.DetailedLibraryTemplateDto;
+import uk.ac.bbsrc.tgac.miso.dto.Dtos;
+import uk.ac.bbsrc.tgac.miso.dto.LibraryTemplateDto;
+import uk.ac.bbsrc.tgac.miso.service.LibraryTemplateService;
+import uk.ac.bbsrc.tgac.miso.webapp.util.BulkCreateTableBackend;
+import uk.ac.bbsrc.tgac.miso.webapp.util.BulkEditTableBackend;
+
+@Controller
+@RequestMapping("/librarytemplate")
+@SessionAttributes("librarytemplate")
+public class EditLibraryTemplateController {
+
+  @Autowired
+  private LibraryTemplateService libraryTemplateService;
+
+  @Value("${miso.detailed.sample.enabled}")
+  private Boolean detailedSample;
+
+  public Boolean isDetailedSampleEnabled() {
+    return detailedSample;
+  }
+
+  private final class BulkCreateLibraryTemplateBackend extends BulkCreateTableBackend<LibraryTemplateDto> {
+
+    public BulkCreateLibraryTemplateBackend(LibraryTemplateDto dto, Integer quantity) throws IOException {
+      super("libraryTemplate", LibraryTemplateDto.class, "Library Templates", dto, quantity);
+    }
+
+    @Override
+    protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) throws IOException {
+      // No config required
+    }
+  }
+
+  private final class BulkEditLibraryTemplateBackend extends BulkEditTableBackend<LibraryTemplate, LibraryTemplateDto> {
+
+    public BulkEditLibraryTemplateBackend() {
+      super("libraryTemplate", LibraryTemplateDto.class, "Library Templates");
+    }
+
+    @Override
+    protected Stream<LibraryTemplate> load(List<Long> ids) throws IOException {
+      return libraryTemplateService.listByIdList(ids).stream();
+    }
+
+    @Override
+    protected LibraryTemplateDto asDto(LibraryTemplate model) {
+      return Dtos.asDto(model);
+    }
+
+    @Override
+    protected void writeConfiguration(ObjectMapper mapper, ObjectNode config) {
+      // No config required
+    }
+  };
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this empty statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AEmptyStatementUsageCheck)